crates/bitfinity-block-validator/src/lib.rs

```diff
@@ -0,0 +1,147 @@
+//! Bitfinity block validator.
+use std::collections::HashSet;
+
+use did::{BlockConfirmationData, BlockConfirmationResult};
+use ethereum_json_rpc_client::{Client, EthJsonRpcClient};
+use eyre::eyre;
+use reth_chain_state::MemoryOverlayStateProvider;
+use reth_evm::execute::{BasicBatchExecutor, BatchExecutor};
+use reth_evm_ethereum::{
+    execute::{EthExecutionStrategy, EthExecutionStrategyFactory},
+    EthEvmConfig,
+};
+use reth_node_types::NodeTypesWithDB;
+use reth_primitives::{Block, BlockWithSenders};
+use reth_provider::{
+    providers::ProviderNodeTypes, ChainSpecProvider as _, ExecutionOutcome,
+    HashedPostStateProvider as _, LatestStateProviderRef, ProviderFactory,
+};
+use reth_revm::{batch::BlockBatchRecord, database::StateProviderDatabase};
+
+/// Block validator for Bitfinity.
+///
+/// The validator validates the block by executing it and then
+/// confirming it on the EVM.
+#[derive(Clone)]
+#[allow(missing_debug_implementations)]
+pub struct BitfinityBlockValidator<C, DB>
```

Comment:
Inside the evm-canister this process is called "Confirmation" to distinguish it from a different already existing process called "Validation". The endpoint is indeed called `send_block_confirmation`. 
Please rename all modules/packages/structs/docs to use this nomenclature

Reply:
Done